openscap_daemon/rest_api.py

```diff
@@ -0,0 +1,370 @@
+# Copyright 2015 Red Hat Inc., Durham, North Carolina.
+# All Rights Reserved.
+#
+# openscap-daemon is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# openscap-daemon is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with openscap-daemon.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#   Mario Vazquez <mavazque@redhat.com>
+
+import threading
+import json
+import os
+
+from datetime import datetime
+from flask import Flask, request
+
+
+class OpenSCAPRestApi(object):
+    """Internal class that implements the REST API using Flask"""
+    def __init__(self, system_instance):
+        super(OpenSCAPRestApi, self).__init__()
+
+        self.app = Flask(__name__)
+        self.system = system_instance
+
+        if self.system.config.api_enabled:
```

Comment:
I would expect this check to be present somewhere else, perhaps before this class is even attempted to be instantiated. That way we can avoid importing this whole module and getting ImportErrors. We should do the same thing for dbus. Only import if it's enabled in config and only import the `dbus` modules inside it.

Reply:
Check moved to oscapd binary. Modified the code just a bit so if REST API is not enabled, flask package won't be imported, hence it will be optional.